tests/test_app.py

```diff
@@ -1,59 +1,176 @@
-        self.assertEqual(response.status_code, 400, response.json)
+        try:
+            # Create the BlobServiceClient object which will be used to create a container client
+            blob_service_client = BlobServiceClient.from_connection_string(FERTISCAN_STORAGE_URL)
 
-    def update_empty_inspection(self):
-        response = test_client.put('/inspections', headers=self.headers)
+            # List all containers to test the connection
+            containers = blob_service_client.list_containers()
+
+            for container in containers:
+                print(container.name)
+
+        except Exception as e:
+            self.fail("Connection failed:", str(e))
+
+    def test_signup_missing_username(self):
+        response = test_client.post(
+            '/signup',
+            headers={ **self.headers,  'Authorization': f'Basic {self.credentials("", self.password)}'},
+            content_type='application/x-www-form-urlencoded',
+        )
         self.assertEqual(response.status_code, 400, response.json)
+    
+    def test_signup(self):
+        response = test_client.post(\
+            '/signup',
+            headers=self.headers,
+            content_type='application/x-www-form-urlencoded',
+        )
+        if response.json['message'] != "User already exists":
+            self.assertEqual(response.status_code, 201, response.json)
 
-    def test_missing_username(self):
-        response = test_client.post('/login', headers=self.headers)
+    def test_login_missing_username(self):
+        response = test_client.post('/login', headers={ **self.headers,  'Authorization': f'Basic {self.credentials("", self.password)}'},
+)
         self.assertEqual(response.status_code, 400, response.json)
     
-    def test_unknow_username(self):
+    def test_login_unknow_username(self):
         username = str(uuid.uuid4())
-        response = test_client.post('/login', data={'username': username, 'password': 'password1'})
+        response = test_client.post(\
+            '/login',
+            headers={ **self.headers,  'Authorization': f'Basic {self.credentials(username, self.password)}'},
+            content_type='application/x-www-form-urlencoded',
+        )
         self.assertEqual(response.status_code, 401, response.json)
 
+    def test_login(self):
+        response = test_client.post(
+            '/login',
+            headers=self.headers,
+        )
+        self.assertEqual(response.status_code, 200, response.json)
+
+    def test_create_empty_inspection(self):
+        response = test_client.post('/inspections', headers=self.headers)
+        self.assertEqual(response.status_code, 500, response.json)
+
+
+    # If the inspection is created, it should at least return a message testifying that.
+    def test_create_inspection(self):
+        with requests.get('https://raw.githubusercontent.com/ai-cfia/fertiscan-pipeline/main/expected.json') as response:
+            json_data = str(response.content)
```

Comment:
Do you mean `json_data = response.json()`?

Reply:
You're right.